### DISCUSSION TITLE
How to get the uninterpolated path from the request?

### DISCUSSION THREAD

**Discussion by kigawas** [fastapi/fastapi#3909] [Questions]
Say I have an endpoint: `/user/{id}`, and access it by `/user/1`. 

My question is how I can get the uninterpolated path `/user/{id}` from the request instead of `/user/1`

**Comment by HarrySky** [comment]
If your endpoints don't have numbers somewhere - following can be done:

```python
import re

endpoint_ids_rgx = re.compile(r"[0-9]+")

# inside endpoint function with "request: Request" argument
path: str = request.scope["path"]
endpoint = endpoint_ids_rgx.sub("{id}", path)
```

It will convert "/your/endpoint/253/action/3" to "/your/endpoint/{id}/action/{id}"

If you actually need original path - it was harder to get, but somebody came up with solution, you need to search for it it either in FastAPI or Starlette issues

**Comment by ghandic** [comment]
Messy - but you can do this

```python
import inspect

from fastapi import FastAPI, APIRouter
import uvicorn

api = FastAPI()
router = APIRouter()


def get_path(name: str):
    for r in router.routes:
        if r.endpoint.__name__ == name:
            return r.path


@router.post("/{id}")
async def example(id: int):
    print(get_path(inspect.currentframe().f_code.co_name))
    ...


api.include_router(router)

```

**Reply by ghandic** [reply]
Note the inspection is just used to get the name of the function. Each time recursing might be wasteful so you might want to store that outside in your own mapping of function name to route

```python
import inspect

from fastapi import FastAPI, APIRouter
import uvicorn

api = FastAPI()
router = APIRouter()
lookup = {}


@router.post("/{id}")
async def example(id: int):
    print(lookup[inspect.currentframe().f_code.co_name])
    ...


api.include_router(router)


for r in router.routes:
    lookup[r.endpoint.__name__] = r.path

```

**Reply by ghandic** [reply]
What is the use-case?
